src/js/main.js

```diff
@@ -64,41 +64,95 @@ Site.Menu = {
 };
 
 Site.Shapes = {
-  currentState: 0,
-  maxState: 7,
+  $mapPattern: $('.map-pattern'),
+  $patterns: [
+    $('#pattern-1'),
+    $('#pattern-2'),
+    $('#pattern-3'),
+    $('#pattern-4'),
+    $('#pattern-5'),
+  ],
+  patternMin: 0,
+  patternMax: 4,
+  currentPattern: 0,
   timer: null,
-  interval: 12000,
+  interval: 100,
+  animating: false,
+
   init: function() {
     var _this = this;
 
-    _this.setState();
-    _this.startInterval();
+    _this.showPattern();
+
+    document.querySelector('#map').addEventListener('startpan', function() {
+      _this.startAnimation();
+    });
+
+    document.querySelector('#map').addEventListener('stoppan', function() {
+      _this.stopAnimation();
+    });
   },
 
-  startInterval: function() {
+  showPattern: function() {
     var _this = this;
 
-    _this.timer = setInterval(function() {_this.setState();}, _this.interval);
+    // Choose 1 or 0
+    var patternStyle = parseInt(Math.random() * 2);
+
+    if (patternStyle === 1) {
+      // If 1, fill paths
+      $('#background-pattern-holder').addClass('fill-path');
+    } else {
+      // If 0, stroke paths
+      $('#background-pattern-holder').addClass('stroke-path');
+    }
+
+    // Assign random current pattern array index
+    _this.currentPattern = Math.floor(Math.random() * (_this.patternMax - _this.patternMin + 1)) + _this.patternMin;
+
+    // Show current pattern
+    _this.$patterns[_this.currentPattern].addClass('show');
   },
 
-  nextState: function() {
+  changePattern: function() {
     var _this = this;
 
-    _this.currentState++;
+    // Check is animating has been stopped; we do this here to fail faster
+    if (!_this.animating) {
```

Comment:
not for this branch but maybe we should write the animation in another object together with the pan animation?